core-primitives/attestation-handler/src/collateral.rs

```diff
@@ -125,7 +126,7 @@ impl SgxQlQveCollateral {
 			return None
 		}
 		let data = &parts[0];
-		let signature = &parts[1][0..parts[1].len() - 3]; // Remove the two last chars that 'close' the json
```

Comment:
After trying out the latest changes I was surprised that it would crash out of the blue. I went through the commits and this change is culprit:
with `-2` the ending `"` character is included as well, the `-3` is needed so that won't be included.

```
full_parts1 = 7cc9acefe152914cd9184ffaa45f907c1b9be0eb03c52ffdd8503214062ff8a0c7fb0f5b005aa7a62be0b9a33cd6ea0607120cd87e3e7fc486a74b4c10ab0086"}
signature_2=7cc9acefe152914...b0086"
signature_3=7cc9acefe152914...b0086
```

Reply:
Hmm, sorry about that 😢 
I just tested again and it works for me. The new call `trim()` should get rid of a potential whitespaces in the end of the string. Do you know what the two characters are in your case?
I will try to implement the suggestion by @clangenb so this might solve the issue anyway.

Reply:
I think it was the ending `\0` in this case as well, should have printed those with debug formatting :disappointed:.